bin/node/src/chain_spec.rs

```diff
@@ -49,128 +45,78 @@ where
 
 #[derive(Clone)]
 pub struct AuthorityKeys {
-    account_id: AccountId,
-    aura_key: AuraId,
-    aleph_key: AlephId,
+    pub account_id: AccountId,
+    pub aura_key: AuraId,
+    pub aleph_key: AlephId,
 }
 
-#[derive(Clone, Copy)]
+#[derive(Debug, StructOpt, Clone)]
 pub struct ChainParams {
-    session_period: u32,
-    millisecs_per_block: u64,
+    /// Pass the chain id.
+    ///
+    /// It can be a predefined one (dev) or an arbitrary chain id passed to the genesis block
+    #[structopt(long, value_name = "CHAIN_SPEC")]
+    pub chain_id: Option<String>,
+
+    /// Specify custom base path.
+    #[structopt(long, short = "d", value_name = "PATH", parse(from_os_str))]
+    pub base_path: Option<PathBuf>,
+
+    #[structopt(long)]
+    pub session_period: Option<u32>,
+
+    #[structopt(long)]
+    pub millisecs_per_block: Option<u64>,
+
+    #[structopt(long)]
+    pub chain_name: Option<String>,
+
+    #[structopt(long)]
+    pub token_symbol: Option<String>,
 }
 
 impl ChainParams {
-    pub fn from_cli(session_period: Option<u32>, millisecs_per_block: Option<u64>) -> Self {
-        ChainParams {
-            session_period: Self::param(
-                "session period",
-                session_period,
-                "SESSION_PERIOD",
-                DEFAULT_SESSION_PERIOD,
-            ),
-            millisecs_per_block: Self::param(
-                "millisecs per block",
-                millisecs_per_block,
-                "MILLISECS_PER_BLOCK",
-                DEFAULT_MILLISECS_PER_BLOCK,
-            ),
+    pub fn chain_id(&self) -> &str {
+        match &self.chain_name {
+            Some(id) => id,
+            None => "a0tnet1",
```

Comment:
All the other defaults are for the devnet and this is for the testnet (right?). If this is on purpose please add a comment as to why.

Reply:
Fixed